### ISSUE TITLE
can't add items to TreeView

### ISSUE THREAD

**Issue by sadguitarius** [supercollider/supercollider#6508]
## Environment

* SuperCollider version: 3.14.0 dev
* Operating system: Windows 11
* Other details (Qt version, audio driver, etc.): Qt 6.8

## Steps to reproduce

These commands both fail. I realized this by trying to open Quarks.gui in 3.14

```supercollider

(
t = TreeView().front;
t.addItem(["hi"]);
)

(
t = TreeView().front;
t.insertItem(0, ["hi"]);
)

```

Here's the error:

```
ERROR: Primitive '_QObject_InvokeMethod' failed.
Failed.
RECEIVER:
Instance of TreeView {    (0000026FC8558EF8, gc=64, fmt=00, flg=00, set=06)
  instance variables [36]
    qObject : RawPointer 0000026F83004870
    finalizer : instance of Finalizer (0000026FC86202C8, size=2, set=1)
    virtualSlots : nil
    wasRemoved : false
    font : nil
    resize : Integer 1
    alpha : Float 1.000000   00000000 3FF00000
    decorator : nil
    layout : nil
    userCanClose : true
    deleteOnClose : true
    action : nil
    mouseDownAction : nil
    mouseUpAction : nil
    mouseEnterAction : nil
    mouseLeaveAction : nil
    mouseMoveAction : nil
    mouseOverAction : nil
    mouseWheelAction : nil
    keyDownAction : nil
    keyUpAction : nil
    keyModifiersChangedAction : nil
    keyTyped : nil
    focusGainedAction : nil
    focusLostAction : nil
    dragLabel : nil
    beginDragAction : nil
    canReceiveDragHandler : nil
    receiveDragHandler : nil
    toFrontAction : nil
    endFrontAction : nil
    onClose : nil
    onResize : nil
    onMove : nil
    itemPressedAction : nil
    onItemChanged : nil
}
CALL STACK:
	MethodError:reportError
		arg this = <instance of PrimitiveFailedError>
	Nil:handleError
		arg this = nil
		arg error = <instance of PrimitiveFailedError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of PrimitiveFailedError>
	Object:throw
		arg this = <instance of PrimitiveFailedError>
	Object:primitiveFailed
		arg this = <instance of TreeView>
	TreeView:addItem
		arg this = <instance of TreeView>
		arg strings = [*1]
	Interpreter:interpretPrintCmdLine
		arg this = <instance of Interpreter>
		var res = nil
		var func = <instance of Function>
		var code = "(
t = TreeView().front;
t.ad..."
		var doc = nil
		var ideClass = <instance of Meta_ScIDE>
	Process:interpretPrintCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Primitive '_QObject_InvokeMethod' failed.
Failed.
RECEIVER: a TreeView
```

**Comment by capital-G** [comment]
I can reproduce this on macOS 15.

**Comment by sadguitarius** [comment]
Looking at the debug output in Visual Studio, I can see an error reading
`QMetaMethod::invoke: cannot convert formal parameter 0 from QtCollider::SafePtr<QcTreeWidget::Item> in call to QcTreeWidget::addItem(QcTreeWidget::ItemPtr,QVariantList)`

**Comment by sadguitarius** [comment]
[this](https://github.com/cutelyst/cutelyst/issues/391) maybe? I'll see if I can get my head around how to try implementing that change here.

**Comment by dyfer** [comment]
To clarify: this issue does not occur on SC 3.13 / qt5, so this seems to be a regression after migrating to Qt6.

**Comment by sadguitarius** [comment]
Looks like it gets to the `mm.invoke` call in the `invokeMethod` method of QObjectProxy with what appear to be the proper arguments and then returns a failure. Could it possibly have something to do with the use of SafePtr?

**Comment by sadguitarius** [comment]
Ok mostly got it working! It was actually that Qt was comparing the literal strings `"QcTreeWidget::ItemPtr"` and `"QtCollider::SafePtr<QcTreeWidget::Item>"` in the `invokeImpl` method in qmetaobject.cpp. I just tried changing `ItemPtr` from a typedef to a compiler definition and this does fix the issue, although not in the most ideal way. Unfortunately, I can't figure out exactly where things are breaking down. I'm sure the typedef could work with a simple variable replacement somewhere, but I've tried a ton of combinations and nothing's working yet. Any ideas?

**Comment by dyfer** [comment]
Thanks so much for working on this, @sadguitarius !
I really have no idea if this is the right way to fix this... I hope someone more familiar with Qt can chime in!

**Comment by sadguitarius** [comment]
My pleasure! I learned some things about Qt along the way, but I started out knowing nothing, so I wouldn't be surprised if there's a better way to handle it. From what I can tell, the use of `QGenericArgument` is not recommended, and if there's a full switch to Qt6, there appear to be some more modern ways of dealing with types and methods, but that would also involve some more intense rewriting. I won't be offended if the fix goes in a different direction :)
